developing-without-a-build/es-dev-server.md

```diff
@@ -0,0 +1,195 @@
+# Developing Without a Build: es-dev-server
+This article is part of a series on developing without a build:
+- Introduction
+- es-dev-server (this article)
+- Testing (coming soon!)
+
+In the introduction, we looked into different workflows and browser APIs. In this article, we will look into how we can set up `es-dev-server`, and how it helps with development without a build.
+
+## es-dev-server
+
+[es-dev-server](https://www.npmjs.com/package/es-dev-server) is a composable web server that focuses on developer productivity when developing with a build. Through options, you can opt into features such as caching, reloading on file changes, SPA routing, resolving bare module imports and compatibility modes to support older browsers.
+
+## Setup
+To start let's create an empty npm project and install `es-dev-server`:
+
+```bash
+npm init
+cd <your-project-name>
+npm i -D es-dev-server
+```
+
+Create an `index.html` in the root of your project:
+```html
+<!DOCTYPE html>
+<html>
+<head></head>
+
+<body>
+  <div id="app"></div>
+  <script type="module" src="./src/app.js"></script>
+</body>
+
+</html>
+```
+
+Create a `src/app.js` file:
+```javascript
+console.log('hello world');
+```
+
+We can now view our application with `es-dev-server`:
+```bash
+npx es-dev-server --open
+```
+
+## Bare imports
+One of the first things you will run into when developing without any build tools is how to deal with bare module imports like this:
+
+```javascript
+import foo from 'foo';
+```
+
+Out of the box, the browser will throw an error, as it doesn't know how to handle these kinds of imports. In our previous article (TODO: Link) we explored how these imports by using import maps, an upcoming browser API.
+
+Until import maps are properly supported in browsers, we can use the `--node-resolve` flag of the dev server. This will rewrite imports in your modules using NodeJS module resolution before they are served to the browser.
+
+To see how this works let's add `lit-html`, a HTML templating library, to our project:
+
+```bash
+npm i -D lit-html
+```
+
+Change `src/app.js` to import it:
+```javascript
+import { render, html } from 'lit-html';
+
+const template = html`<p>Hello world!</p>`;
+render(template, document.getElementById('app'));
+```
+
+Now let's restart our server, adding the node resolve flag:
+```bash
+npx es-dev-server --node-resolve --open
```

Comment:
i'd say throughout the blog either go with the installed version or npx (see comment above as well)